src/Uno.UWP/Storage/Streams/DataReader.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Windows.Storage.Streams
+{
+	public partial class DataReader : IDataReader, IDisposable
+	{
+		private readonly IBuffer _buffer = null;
+		private uint _position = 0;
+
+		public ByteOrder ByteOrder { get; set; }
+		public UnicodeEncoding UnicodeEncoding { get; set; }
+
+		public uint UnconsumedBufferLength
+		{
+			get => _buffer.Length - _position;
+		}
+
+		private void CheckPosition(uint size)
+		{
+			if (_buffer.Length >= (_position + size))
+			{
+				return;
+			}
+
+			// "internal index" , not parameter error
+			throw new IndexOutOfRangeException("Windows.Storage.Streams.DataReader - reading past EOF");
+		}
+
+		public byte ReadByte()
+		{
+			CheckPosition(1);
+			byte value = _buffer.Data[_position];
+			_position++;
+			return value;
+		}
+
+		public void ReadBytes(byte[] value)
+		{
+			CheckPosition((uint)value.Length);

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'value' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5527039)